nexus/chart/templates/deploymentconfig.yml

```diff
@@ -0,0 +1,99 @@
+kind: DeploymentConfig
+apiVersion: apps.openshift.io/v1
+metadata:
+  name: {{ .Values.global.appName }}
+  labels:
+    app: {{ .Values.global.appName }}
+spec:
+  strategy:
+    type: Recreate
+    recreateParams:
+      timeoutSeconds: 600
+    resources: {}
+    activeDeadlineSeconds: 21600
+  triggers:
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - {{ .Values.global.appName }}
+        from:
+          kind: ImageStreamTag
+          name: {{ .Values.global.appName }}:{{ .Values.global.odsImageTag }}
+          namespace: {{ .Values.global.odsNamespace }}
+      type: ImageChange
+    - type: ConfigChange
+  replicas: 1
+  revisionHistoryLimit: 10
+  test: false
+  selector:
+    app: {{ .Values.global.appName }}
+    deploymentconfig: {{ .Values.global.appName }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.global.appName }}
+        deploymentconfig: {{ .Values.global.appName }}
+    spec:
+      volumes:
+        - name: {{ .Values.global.appName }}-data
+          persistentVolumeClaim:
+            claimName: {{ .Values.global.appName }}-data
+        - name: {{ .Values.global.appName }}-db-backup
+          persistentVolumeClaim:
+            claimName: {{ .Values.global.appName }}-db-backup
+      containers:
+        - resources:
+            limits:
+              cpu: {{ .Values.nexus.cpuLimit }}
+              memory: {{ .Values.nexus.memLimit }}
+            requests:
+              cpu: {{ .Values.nexus.cpuRequest }}
+              memory: {{ .Values.nexus.memRequest }}
+          terminationMessagePath: /dev/termination-log
+          name: {{ .Values.global.appName }}
+          livenessProbe:
+            httpGet:
+              path: /service/rest/v1/status/writable
+              port: 8081
+              scheme: HTTP
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 2
+            failureThreshold: 3
+          readinessProbe:
+            httpGet:
+              path: /service/rest/v1/status/writable
+              port: 8081
+              scheme: HTTP
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 2
+            failureThreshold: 3
```

Comment:
I think defining both liveness and readiness probe exactly the same way isn't ideal. The readiness probe could be removed as you restart the pod anyway at the same time it would be marked as not ready. Maybe consider to remove the readiness probe, or allow a higher failure threshold for the liveness probe, see https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#:~:text=A%20common%20pattern%20for%20liveness,before%20it%20is%20hard%20killed..

Reply:
I increase it to 10 the failure rate for liveness as I would preffer to maintain the readiness probe